chore: Sync markdown to DITA, volume 2 (#1255)
kyma-project/telemetry-manager: docs/user/02-logs.md

With application logs, you can debug an application and derive the internal state of an application. When logs are emitted with the correct severity level and context, they're essential for observing an application.

## Overview

The Telemetry module provides the [Fluent Bit](https://fluentbit.io/) log agent for the collection and shipment of application logs of any container running in the Kyma runtime.

You can configure the log agent with external systems using runtime configuration with a dedicated Kubernetes API (CRD) named `LogPipeline`. With the LogPipeline's HTTP output, you can natively integrate with vendors that support this output, or with any vendor using a [Fluentd integration](https://medium.com/hepsiburadatech/fluent-logging-architecture-fluent-bit-fluentd-elasticsearch-ca4a898e28aa).

<!--- custom output/unsupported mode is not part of Help Portal docs --->
If you want more flexibility than provided by the proprietary protocol, you can run the agent in the [unsupported mode](#unsupported-mode), using the full vendor-specific output options of Fluent Bit. If you need advanced configuration options, you can also bring your own log agent.

## Prerequisites

> Your application must log to `stdout` or `stderr`, , which ensures that the logs can be processed by Kubernetes primitives like `kubectl logs`. For details, see [Kubernetes: Logging Architecture](https://kubernetes.io/docs/concepts/cluster-administration/logging/).

Reviewer: ```suggestion
Your application must log to `stdout` or `stderr`, which ensures that the logs can be processed by Kubernetes primitives like `kubectl logs`. For details, see [Kubernetes: Logging Architecture](https://kubernetes.io/docs/concepts/cluster-administration/logging/).
```

## Architecture

### Log Agent

[Fluent Bit](https://fluentbit.io/), as a log agent, collects all application logs of the cluster workload and ships them to a backend.

![Architecture](./assets/logs-arch.drawio.svg)

1. Container logs are stored by the Kubernetes container runtime under the `var/log` directory and its subdirectories.

> 2. Fluent Bit runs as a [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) (one instance per Node), detects any new log files in the folder, and tails them using a filesystem buffer for reliability.

Author: thanks! I added it :)

Kyma's Telemetry module brings a predefined setup of the Fluent Bit DaemonSet and a base configuration, which assures that the application logs of the workloads in the cluster are processed reliably and efficiently. Additionally, the Telemetry module provides a Kubernetes API called `LogPipeline` to configure outputs with some filtering capabilities.

This approach ensures reliable buffer management and isolation of pipelines, while keeping flexibility on customizations.

![Pipeline Concept](./assets/logs-pipelines.drawio.svg)

1. A dedicated `tail` **input** plugin reads the application logs, which are selected in the input section of the `LogPipeline`. Each `tail` input uses a dedicated `tag` with the name `<logpipeline>.*`.

2. The application logs are enriched by the `kubernetes` **filter**. You can add your own filters to the default filters.

3. Based on the default and custom filters, you get the desired **output** for each `LogPipeline`.

This approach assures a reliable buffer management and isolation of pipelines, while keeping flexibility on customizations.

### Telemetry Manager

> The LogPipeline resource is resource is watched by Telemetry Manager, which is responsible for generating the custom parts of the Fluent Bit configuration.

Reviewer: ```suggestion
The LogPipeline resource is watched by Telemetry Manager, which is responsible for generating the custom parts of the Fluent Bit configuration.
```

![Manager resources](./assets/logs-resources.drawio.svg)

- Telemetry Manager watches all LogPipeline resources and related Secrets.
- Whenever the configuration changes, Telemetry Manager validates the configuration (with a [validating webhook](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/)) and generates a new configuration for the Fluent Bit DaemonSet, where several ConfigMaps for the different aspects of the configuration are generated.
- Furthermore, referenced Secrets are copied into one Secret that is also mounted to the DaemonSet.

## Setting up a LogPipeline

In the following steps, you can see how to construct and deploy a typical LogPipeline. Learn more about the available [parameters and attributes](resources/02-logpipeline.md).

> ### Step 1: Create a LogPipeline and Output

Author: I'd rather remove the word "step". Thanks for spotting this inconsistency! 
Fixing this.

### Step 4: Rotate the Secret

Telemetry Manager continuously watches the Secret referenced with the **secretKeyRef** construct. You can update the Secret’s values, and Telemetry Manager detects the changes and applies the new Secret to the setup.

> [!TIP]
> If you use a Secret owned by the [SAP BTP Operator](https://github.com/SAP/sap-btp-service-operator), you can configure an automated rotation using a `credentialsRotationPolicy` with a specific `rotationFrequency` and don’t have to intervene manually.

### Step 5: Deploy the Pipeline

To activate the constructed LogPipeline, apply the  `logpipeline.yaml` resource file in your cluster:

```bash
kubectl apply -f logpipeline.yaml
```

### Result

You activated a LogPipeline and logs start streaming to your backend.

> To check that the pipeline is running, verify that the status of the LogPipeline in your cluster is `READY`:

Reviewer 2: ```suggestion
To check that the pipeline is running, wait until all status conditions of the LogPipeline in your cluster have status `True`:
```

## Operations

> The Telemetry module ensures that the log agent instances are operational and healthy at any time, for example, with buffering and retries. However, there may be situations when the instances drop logs, or cannot handle the log load.

Author: I missed syncing this. Doing it now!

- **Buffer Limits**: Fluent Bit buffers up to 1 GB of logs if a configured output cannot receive logs. The oldest logs are dropped when the limit is reached or after 300 retries.
- **Throughput**: Each Fluent Bit Pod (each running on a dedicated Node) can process up to 10 MB/s of logs for a single LogPipeline. With multiple pipelines, the throughput per pipeline is reduced. The used logging backend or performance characteristics of the output plugin might limit the throughput earlier.
- **Max Amount of Pipelines**: The maximum amount of LogPipeline resources is 5.

### Unsupported Mode
<!--- unsupported mode is not part of Help Portal docs --->
The `unsupportedMode` attribute of a LogPipeline indicates that you are using a `custom` filter and/or `custom` output. The Kyma team does not provide support for a custom configuration.

### Fluent Bit Plugins
<!--- Fluent Bit Plugins is not part of Help Portal docs --->

You cannot enable the following plugins, because they potentially harm the stability:

- Kubernetes Filter
- Rewrite_Tag Filter

## Troubleshooting

### No Logs Arrive at the Backend

> **Symptom**: No logs arrive at the backend. In the LogPipeline status, the `TelemetryFlowHealthy` condition has status **AllDataDropped**.

Reviewer: In the guide below you use a bullet list for the two symptoms, I prefer it here as well

## Limitations

> 

Author: in DITA, it looks better as a list instead of subsection headlines

- **Reserved Log Attributes**: The log attribute named `kubernetes` is a special attribute that's enriched by the `kubernetes` filter. When you use that attribute as part of your structured log payload, the metadata enriched by the filter are overwritten by the payload data. Filters that rely on the original metadata might no longer work as expected.
- **Buffer Limits**: Fluent Bit buffers up to 1 GB of logs if a configured output cannot receive logs. The oldest logs are dropped when the limit is reached or after 300 retries.
- **Throughput**: Each Fluent Bit Pod (each running on a dedicated Node) can process up to 10 MB/s of logs for a single LogPipeline. With multiple pipelines, the throughput per pipeline is reduced. The used logging backend or performance characteristics of the output plugin might limit the throughput earlier.
- **Max Amount of Pipelines**: The maximum amount of LogPipeline resources is 5.

> ### Unsupported Mode

Author: in markdown, I need this subheadline so that the links to this section work
